src/hooks/useLocations.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+
+export const useLocations = (lat, lng) => {
+  const [locations, setLocations] = useState([]);
+  const [status, setStatus] = useState('idle');
+  const [error, setError] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+
+  useEffect(() => {
+    // Abort fetch
+    let abortController = new AbortController();
```

Comment:
And also TIL about new built-in JS objects!

Reply:
Also, I learned `AbortController` with this task. Yes, you're right about your comments above, also I would like to comment that it could be useful to 'cancel' calls to APIs when you're component is unmounted with a 'clean-up' function.